bugshaker/src/main/java/com/github/stkent/bugshaker/MainActivity.java

```diff
@@ -0,0 +1,232 @@
+package com.github.stkent.bugshaker;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+
+import android.app.Application;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.BitmapDrawable;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.Toast;
+
+import com.github.stkent.bugshaker.flow.email.FeedbackEmailFlowManager;
+import com.github.stkent.bugshaker.flow.widget.DrawingView;
+import com.github.stkent.bugshaker.utilities.LogcatUtil;
+import com.github.stkent.bugshaker.utilities.Logger;
+import com.github.stkent.bugshaker.utilities.SendEmailUtil;
+
+public class MainActivity extends AppCompatActivity implements OnClickListener {
+
+	private DrawingView drawView;
+
+	private ImageButton currPaint;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_main);
+
+		drawView = (DrawingView)findViewById(R.id.drawing);
+		LinearLayout paintLayout = (LinearLayout)findViewById(R.id.paint_colors);
+		currPaint = (ImageButton)paintLayout.getChildAt(0);
+		currPaint.setImageDrawable(getResources().getDrawable(R.drawable.paint_pressed));
+
+		ImageButton drawBtn = (ImageButton) findViewById(R.id.draw_btn);
+		drawBtn.setOnClickListener(this);
+
+		drawView.setBrushSize(getResources().getInteger(R.integer.small_size));
+
+		ImageButton eraseBtn = (ImageButton) findViewById(R.id.erase_btn);
+		eraseBtn.setOnClickListener(this);
+
+		ImageButton sendBtn = (ImageButton) findViewById(R.id.sendEmail);
+		sendBtn.setOnClickListener(this);
+
+		ImageButton speechBtn = (ImageButton) findViewById(R.id.speechBox);
+		speechBtn.setOnClickListener(this);
+
+
+		Logger logger = new Logger(true);
+		Application application = getApplication();
+
+		String pathOfScreenshot = getIntent().getStringExtra("uri");
+		File screenshotFile = new File(pathOfScreenshot);
+
+		if(screenshotFile.exists()) {
+			Bitmap myBitmap = BitmapFactory.decodeFile(pathOfScreenshot);
+
+			drawView = (DrawingView) findViewById(R.id.drawing);
+			Drawable temp = new BitmapDrawable(getResources(), myBitmap);
+			drawView.setBackgroundDrawable(temp);
+		}
+		else {
+			Log.d("MainActivity", "File does not exist");
+		}
+	}
+
+	public void setButtonClickListener(String string, ImageButton button, final Dialog dialog, boolean isMarker) {
+		String str = string;
```

Comment:
maybe have the size passed into this method? Then we don't need this "string" parameter.